core/web/log_controller.go

```diff
@@ -0,0 +1,45 @@
+package web
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/chainlink"
+	"github.com/smartcontractkit/chainlink/core/web/presenters"
+	"go.uber.org/zap/zapcore"
+)
+
+// LogController manages the logger config
+type LogController struct {
+	App chainlink.Application
+}
+
+type loglevelPatchRequest struct {
+	EnableDebugLog *bool `json:"debugEnabled"`
+}
+
+// ToggleDebug toggles the debug log mode
+func (cc *LogController) ToggleDebug(c *gin.Context) {
+	request := &loglevelPatchRequest{}
+	if err := c.ShouldBindJSON(request); err != nil {
+		jsonAPIError(c, http.StatusUnprocessableEntity, err)
+		return
+	}
+
+	if *request.EnableDebugLog {
+		cc.App.GetStore().Config.Set("LOG_LEVEL", zapcore.DebugLevel.String())
+	} else {
+		cc.App.GetStore().Config.Set("LOG_LEVEL", zapcore.InfoLevel.String())
+	}
+	logger.SetLogger(cc.App.GetStore().Config.CreateProductionLogger())
```

Comment:
This logging configuration will not persist across restarts. Do you know if it is a requirement that it does so?

If you do need to, you will need to persist the config into the DB